src/Microsoft.Health.Fhir.Core/Models/PartialDateTime.cs

```diff
@@ -292,7 +301,7 @@ public static PartialDateTime Parse(string s)
             catch (Exception ex) when (ex is ArgumentException)
             {
                 // The input value was parsed correctly but one of the value provided were out of range.
-                throw new FormatException("Input string was not in a correct format. At least one portion of a date was invalid or out of range.", ex);
+                throw new RequestNotValidException("Input string was not in a correct format. At least one portion of a date was invalid or out of range: " + ex.Message);
```

Comment:
Include ex as the inner exception

Reply:
This type of exception doesn't have a constructor that takes an inner exception.

Reply:
Refactored to remove outer exception.